lib/ssh_scan/config_loader/yaml.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+
+module SSHScan
+  module ConfigLoader
+    class Yaml
+      def initialize(config, class_name = nil, mode = 'file', logger = nil)
+        @opts = case mode
+          when "file" then YAML.load_file(config)
+          when "string" then YAML.load(config)
+        end
```

Comment:
I think we should have config loaded take a similar abstraction strategy as we did with the Database abstractions.  ConfigLoader should be able to be pointed as a file and figure it out, but we should have subclasses that do the heavy lifting (if any) for that particular config type format (JSON, YML, etc.)